src/Connection/Http1Connection.php

```diff
@@ -297,9 +297,8 @@ private function readResponse(
                 $status = $response->getStatus();
 
                 if ($status === Http\HttpStatus::SWITCHING_PROTOCOLS) {
-                    $connection = Http\parseMultipleHeaderFields($response, 'connection')[0] ?? null;
-
-                    if (!isset($connection['upgrade'])) {
+                    $connection = array_map('strtolower', Http\splitHeader($response, 'connection'));
+                    if (!\in_array('upgrade', $connection, true)) {
```

Comment:
@trowski We should probably make this simpler with an API provided by `amphp/http`.

Reply:
How about
```php
$connection = Http\parseMultipleHeaderFields($response, 'connection')[0] ?? [];
if (!\array_key_exists('upgrade', $connection)) {
    // ...
}
```

Reply:
@rekryt The issue here is the casing, I think.

Reply:
It might be both actually. Maybe keys without a value should have an empty string as the value.

Reply:
Keys without a value should have an "null" as value.
```
$connection = [ 'upgrade' => null ];
isset($connection['upgrade'] === false;
```

Reply:
Updated to the newly introduced `parseHeaderTokens`.